### ISSUE TITLE
Grid generation fails with dimensionally disparate npoints

### ISSUE THREAD

**Issue by MTCam** [inducer/meshmode#74]
It would be good if we could make rectangular grids with uniform (or arbitrary) spacing.   The example here is a rectangle [0, 10] x [0 , 1]  wherein uniform spacing is attempted, i.e. npts = (101, 11).   The grid generator appears to fail to produce this grid.

```Python
x0 = 0.0
y0 = 0.0
x1 = 10.0
y1 = 1.0
nx = 101
ny = 11

npts = (nx, ny)
a = (x0, y0)
b = (x1, y1)

from meshmode.mesh.generation import generate_regular_rect_mesh
mesh = generate_regular_rect_mesh(a=a, b=b, n=npts)
```

**Comment by inducer** [comment]
Wasn't this fixed by #72?

**Comment by MTCam** [comment]
Fixes so fast you'll freak?   Holy cow.
